packer/builder/azure/driver_restapi/config.go

```diff
@@ -0,0 +1,200 @@
+package driver_restapi
+
+import (
+	"fmt"
+	"github.com/MSOpenTech/packer-azure/packer/builder/azure/driver_restapi/targets"
+	"github.com/MSOpenTech/packer-azure/packer/builder/azure/utils"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	"log"
+	"os"
+	"regexp"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+
+	SubscriptionName        string `mapstructure:"subscription_name"`
+	PublishSettingsPath     string `mapstructure:"publish_settings_path"`
+	StorageAccount          string `mapstructure:"storage_account"`
+	StorageAccountContainer string `mapstructure:"storage_account_container"`
+	OsType                  string `mapstructure:"os_type"`
+	OsImageLabel            string `mapstructure:"os_image_label"`
+	Location                string `mapstructure:"location"`
+	InstanceSize            string `mapstructure:"instance_size"`
+	UserImageLabel          string `mapstructure:"user_image_label"`
+
+	username         string `mapstructure:"username"`
+	tmpVmName        string
+	tmpServiceName   string
+	tmpContainerName string
+	userImageName    string
+}
+
+func newConfig(raws ...interface{}) (*Config, []string, error) {
+	var c Config
+
+	var md mapstructure.Metadata
+	err := config.Decode(&c, &config.DecodeOpts{
+		Metadata:    &md,
+		Interpolate: true,
+	}, raws...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Defaults
+	log.Println(fmt.Sprintf("%s: %v", "PackerUserVars", c.PackerConfig.PackerUserVars))
+
+	if c.StorageAccountContainer == "" {
+		c.StorageAccountContainer = "vhds"
+	}
+
+	// Check values
+	var errs *packer.MultiError
+
+	if c.SubscriptionName == "" {
+		errs = packer.MultiErrorAppend(errs,
+			fmt.Errorf("subscription_name must be specified"))
+	}
+
+	if c.PublishSettingsPath == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path must be specified"))
+	}
+
+	if c.StorageAccount == "" {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("storage_account must be specified"))
+	}
+
+	if _, err := os.Stat(c.PublishSettingsPath); err != nil {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("publish_settings_path is not a valid path: %s", err))
+	}
+
+	osTypeIsValid := false
+	osTypeArr := []string{
+		targets.Linux,
+		targets.Windows,
+	}
+
+	for _, osType := range osTypeArr {
```

Comment:
loop seems redundant. why not 

    if !(osType == targets.Linux || osType == targets.Windows) already